lib/spack/spack/target.py

```diff
@@ -102,7 +102,7 @@ def to_dict_or_value(self):
         if self.microarchitecture.vendor == "generic":
             return str(self)
 
-        return syaml.syaml_dict(self.microarchitecture.to_dict(return_list_of_items=True))
+        return syaml.syaml_dict(self.microarchitecture.to_dict().items())
```

Comment:
This is a relic from Python 2.7, so it has been changed here. Dictionaries are ordered from Python 3.6 on.